### DISCUSSION TITLE
Stride and the Community Toolkit is amazing! Simple, lean, to the point. Stride seems far better in principle than the competition ... Great job all around

### DISCUSSION THREAD

**Discussion by jonmdev** [stride3d/stride#2199] [Contributors]
Like many here probably, I am a Unity refugee. Since October, I have been occupied with other projects (Maui primarily, which I am now trying to integrate with Stride...), while also been seeking a new game engine in general.

## The Competition

To review, from my perspective:

- **Unity**: Good features but ridiculously bloated Editor you are tied to, no native UI systems access, crappy new license, unstable future given that.
- **Unreal**: Good if you're making a massive AAA game but not necessary for small or mobile type projects, percentage revenue cut license.
- **Flax**: Same percentage licensing cut as Unity and Unreal that I think should be ideally avoided at all costs.
- **Godot**: Good for 2D games perhaps and some 3D but no practical code only option or native UI integration.
- **Evergine**: Great native UI integration with Maui (they have a simple Maui template in their Editor) but then you are tied to their heavy Editor, unclear stability for future, charges for tech support or code access, bugs seem ignored otherwise, and unknown future license conditions so could end up like Unity.
- **Monogame**: Seems very awkward to get into at least from what I can tell.

My expectation is that Godot and Stride will be the main beneficiaries of Unity's stupidity. 

For 3D projects, I think Stride appears to be certainly the best, especially now that you guys are up to date on .NET 8.0 and theoretically things like Maui integration to beat even Evergine at this should be possible.

## Documentation
You guys have done a fantastic job on the documentation. It makes it so easy to get into.

Your [Unity to Stride explanation](https://doc.stride3d.net/4.0/en/Manual/stride-for-unity-developers/index.html) was top notch.

Your instructions for creating a [code only project](https://stride3d.github.io/stride-community-toolkit/manual/code-only/create-project.html) were also fantastic. Simple, clear, to the point.

Coming from the ridiculous Unity Editor and versioning and bloated projects required by Unity and other game development programs this is like an absolute breeze.

## Code Only vs. Editor Philosophy

I am curious for anyone else's perspective, but from mine, the way I see things ideally now and why I like code-only so much is:

1) Running your game from Code Only (if you can) keeps your code more platform agnostic, makes it easier to track all your objects, and makes it easier to coordinate object interactions (as everything can be instantiated or have references in a few manager scripts).
2) One can use the Editor just to perform visual tests or prepare your assets and prefabs separately, and then pass them into your code only project for instantiation there.

Moving from Unity to other systems was not as catastrophic for me as it could have been in part because I was already managing most of my systems code only in C#. If you instantiate and manage your objects in code already, or programmatically instantiate and move your environments and enemies, etc. then even if you have to change systems again down the line, as long as you can stay same language (eg. C#) it is not so bad.

For example, I had done a lot in UI Toolkit programmatically in Unity and I was able to convert a lot of this to Maui with relative ease (some I rewrote because of new ideas) because again, C# > C# is not hard and you just have to rename or move around some classes to get it done. I was able to reuse thousands of lines of code with almost no changes.

If you develop by contrast entirely in the Editor, then you will be stuck in that Editor. It can become a cobweb of objects and hierarchies.

I see people doing this on GameDev channels on YouTube and I always think these people are boxing themselves into a corner. Their projects look insane. Making a small change in code is easy. Re-manually creating an entire massive Editor hierarchy is nuts. 

It perhaps requires a bit more visualization and conceptualization to do things mostly code only, but I think the benefits are very clear.

## Conclusion

I just wanted to share this as a thank you to the team so far. I think you guys have done a great job so far. I hope the project stays alive and well and keeps remaining current going forward with .NET. If I am able to successfully integrate Maui that I think will be another benefit to Stride. 

If I am successful with my endeavors, for whatever it's worth, I will be happy to donate to keep it alive.

Thanks.

**Comment by VaclavElias** [comment]
Thank you for your comments. Our toolkit is currently in preview, and we are working to merge and unify all the outstanding work from contributors who have created Stride libraries, examples, and experiments in their repositories into the toolkit helpers and extensions.

Both code-only and Editor approaches should be able to utilize these extensions and helpers.

Regarding the code-only approach, I have created a section, [Why Opt for Code-Only Over the Stride Editor](https://stride3d.github.io/stride-community-toolkit/manual/code-only/index.html), where I've listed some reasons. These mainly reflect my perspective and those from a non-game-dev viewpoint. I will definitely add your insights as well, as they are particularly relevant to game developers.

We are happy to provide assistance here or on Discord.

**Comment by Doprez** [comment]
Happy to hear you like the idea of this project. 

The big benefit of Stride compared to Monogame for me was that Monogame seemed like it had a lot of resources for 2D related projects but it was a struggle to find anything 3D related so Strides OOB focus on 3D finally sold me. The editor was a huge bonus for me since I do like it for editing scenes visually. Coming from Unity/Godot and primarily being a web dev, mostly in .NET, Stride just felt like home in a lot of ways.

I have also been a big fan of handling my own data anywhere possible, using SQLite and JSON files. I fully agree with your reasoning for doing as much as possible with a code only approach. Being locked down is something I heavily avoided and it made trying out other engines very easy. Before settling on Stride I think I tried the ones you mentioned above as well as NeoAxis but it had the same issues as Flax in licensing.

Strides current problem is manpower. Its done well IMO, especially since everyone involved in contributions and the project as a whole is a volunteer. As you have found in the first attempts to get the Android/IOS builds working there are still some problems. Most of the consistant people hang around to help with Linux and Windows but Mobile needs some love . 

The current mobile implementation is based on Xamarin IIRC which needs to be updated and I think a good alternative could be MAUI or even Avalonia since there is already ongoing work to port the editor over to Avalonia it may be the favoured choice in the near future.

**Comment by jonmdev** [comment]
Unfortunately, it looks like it will take too much work to integrate with Maui the deeper I look into it. I was able to find another solution that worked faster. Best of luck to the Stride project.
